src/lib.rs

```diff
@@ -101,53 +106,101 @@ pub fn version() -> u32 { unsafe { ffi::bitcoinconsensus_version() as u32 } }
+    let flags = match spent_outputs {
+        Some(_) => VERIFY_ALL_PRE_TAPROOT | VERIFY_TAPROOT,
+        None => VERIFY_ALL_PRE_TAPROOT,
+    };
+
+    verify_with_flags(spent_output, amount, spending_transaction, spent_outputs, input_index, flags)
 }
 
 /// Same as verify but with flags that turn past soft fork features on or off.
 pub fn verify_with_flags(
     spent_output_script: &[u8],
     amount: u64,
     spending_transaction: &[u8],
+    spent_outputs: Option<&[Utxo]>,
     input_index: usize,
     flags: u32,
 ) -> Result<(), Error> {
-    unsafe {
-        let mut error = Error::ERR_SCRIPT;
+    match spent_outputs {
+        Some(spent_outputs) => unsafe {
+            let mut error = Error::ERR_SCRIPT;
 
-        let ret = ffi::bitcoinconsensus_verify_script_with_amount(
-            spent_output_script.as_ptr(),
-            spent_output_script.len() as c_uint,
-            amount,
-            spending_transaction.as_ptr(),
-            spending_transaction.len() as c_uint,
-            input_index as c_uint,
-            flags as c_uint,
-            &mut error,
-        );
-        if ret != 1 {
-            Err(error)
-        } else {
-            Ok(())
-        }
+            let ret = ffi::bitcoinconsensus_verify_script_with_spent_outputs(
+                spent_output_script.as_ptr(),
+                spent_output_script.len() as c_uint,
+                amount,
+                spending_transaction.as_ptr(),
+                spending_transaction.len() as c_uint,
+                spent_outputs.as_ptr() as *const c_uchar,
+                spent_outputs.len() as c_uint,
+                input_index as c_uint,
+                flags as c_uint,
+                &mut error,
+            );
+            if ret != 1 {
+                Err(error)
+            } else {
+                Ok(())
+            }
+        },
+        None => unsafe {
+            let mut error = Error::ERR_SCRIPT;
+
+            let ret = ffi::bitcoinconsensus_verify_script_with_amount(
+                spent_output_script.as_ptr(),
+                spent_output_script.len() as c_uint,
+                amount,
+                spending_transaction.as_ptr(),
+                spending_transaction.len() as c_uint,
+                input_index as c_uint,
+                flags as c_uint,
+                &mut error,
+            );
+            if ret != 1 {
+                Err(error)
+            } else {
+                Ok(())
+            }
+        },
     }
 }
 
+/// Mimics the Bitcoin Core UTXO typedef (bitcoinconsenus.h)
+// This is the TxOut data for utxos being spent, i.e., previous outputs.
+#[repr(C)]
+pub struct Utxo {
+    /// WARNING: This should be the consensus encoded script __without__ the leading var int!
+    ///
+    /// ScriptPubkey's in rust-bitcoin are consensus encoded with a variable byte length value
+    /// prepended to the front - remove the varint then take a pointer to the encoded vector.
```

Comment:
In 33022e2be7c86f81f933d62bd34ccfb9b434b11f:

I'm ok merging this for now and doing a followup PR (or another release) to update the comment. But this comment seems like it's way overcomplicated things. It looks like you just need a raw pointer to the underlying bytes of the `Script` as well as its length. You should be able to just use `[..]` to get a byteslice from a `Script` (or whatever our accessor is for this) then extract the pointer/length from there.

I don't see why you would need to encode and tweak the encoding.